src/_pytest/terminal.py

```diff
@@ -309,6 +310,7 @@ def get_location(self, config: Config) -> Optional[str]:
         return None
 
 
+@final
 class TerminalReporter:
```

Comment:
`TerminalReporter` is subclassed by both `pytest-xdist` and `pytest-sugar`. Not that this class is meant for subclassing, in fact we have broken `pytest-sugar` in the past a few times because of side effects of some refactorings, but I guess this makes it official. 😁

Reply:
I don't see it in `pytest-xdist`, but might be missing it.

As for pytest-sugar, it seems pretty clear to me that it is using private API. We can endeavor to keep it working but I don't think we should guarantee it. So I think the `@final` is good especially for these cases (making it official as you say).

Reply:
> I don't see it in pytest-xdist, but might be missing it.

Ahh sorry, I misremembered, it uses it by aggregation:

https://github.com/pytest-dev/pytest-xdist/blob/master/src/xdist/dsession.py#L331

But unfortunately our own `TerminalReporter` knows/handles xdist itself, hehehe.

> As for pytest-sugar, it seems pretty clear to me that it is using private API. We can endeavor to keep it working but I don't think we should guarantee it. So I think the @final is good especially for these cases (making it official as you say).

Completely agree. I just mentioned those for that effect: to mark it as officially non-supported.